integration-tests/src/merge_query_params.test.js

```diff
@@ -37,3 +37,58 @@ describe("for an endpoint with multiple queryString middleware", () => {
     expect(fetch).toHaveBeenLastCalledWith(expectedURL);
   });
 });
+
+const testOptionalEndpoint: SafeAPI.Endpoint<
+  {
+    x: ?string
+  },
+  string
+> = SafeAPI.endpoint()
+  .fragment("foo")
+  .queryParams({
+    x: T.maybeString
+  });
+
+const testOptionalHandler = async input => {
+  return input.x ? input.x.toString() : "";
+};
+
+describe("for an endpoint with optional queryString middleware", () => {
+  it("can be called with a query parameter", async () => {
+    const server = TestUtils.makeServer({
+      endpoint: testOptionalEndpoint,
+      handler: testOptionalHandler
+    });
+    const baseURL = `http://localhost:${server.address().port}`;
+    const input = { x: "X" };
+    await SafeFetch.safeGet(baseURL, testOptionalEndpoint, input);
+    const expectedURL = `${baseURL}/foo?x=X`;
+    expect(fetch).toHaveBeenLastCalledWith(expectedURL);
+  });
+
+  describe("for null parameters", () => {
+    it("will serialize null parameters to the empty string", async () => {
+      const server = TestUtils.makeServer({
+        endpoint: testOptionalEndpoint,
+        handler: testOptionalHandler
+      });
+      const baseURL = `http://localhost:${server.address().port}`;
+      const input = { x: null };
+      await SafeFetch.safeGet(baseURL, testOptionalEndpoint, input);
+      const expectedURL = `${baseURL}/foo?x=`;
+      expect(fetch).toHaveBeenLastCalledWith(expectedURL);
+    });
+
+    it("will remove undefined parameters", async () => {
+      const server = TestUtils.makeServer({
+        endpoint: testOptionalEndpoint,
+        handler: testOptionalHandler
+      });
+      const baseURL = `http://localhost:${server.address().port}`;
+      const input = { x: undefined };
+      await SafeFetch.safeGet(baseURL, testOptionalEndpoint, input);
+      const expectedURL = `${baseURL}/foo?`;
```

Comment:
should the expectedURL contain the `?` at all? Fine either way.

Reply:
so the query string builder outputs a `?` right now. I'm not against changing it, but this is a valid url.

Reply:
See my comment (I thought I posted it the other day but I see I didn't press submit). Moving the conditional should fix this. It's better to not output the `?` when there are no query params I think.